components/script/dom/readablestreamdefaultreader.rs

```diff
@@ -169,6 +67,19 @@ pub(crate) enum ReadRequest {
     DefaultTee {
         tee_read_request: Dom<DefaultTeeReadRequest>,
     },
+    /// Spec read loop variant, driven by read-request steps (no Promise).
+    /// <https://streams.spec.whatwg.org/#read-loop>
+    ReadLoop {
+        #[ignore_malloc_size_of = "Rc is hard"]
+        #[no_trace]
+        success_steps: Rc<ReadAllBytesSuccessSteps>,
+        #[ignore_malloc_size_of = "Rc is hard"]
+        #[no_trace]
+        failure_steps: Rc<ReadAllBytesFailureSteps>,
+        reader: Dom<ReadableStreamDefaultReader>,
+        #[ignore_malloc_size_of = "Rc is hard"]
```

Comment:
Nit: for `Rc` we can now do `conditional_malloc_size_of`. For `Fn` that doesn't work, but for `DomRefCell` that should work